### PR TITLE
Improve the HTTP client SPI

### PR DESCRIPTION
**Pull request by Ladicek**
The main changes are:
- move `RegistryHttpClient` to the `spi` package, so that all
  HTTP client SPI classes are there
- renamed `RestClientProvider`, which is a provider
  of `RegistryHttpClient` instances, to `RegistryHttpClientProvider`
- renamed `RestClientServiceLoader`, which uses `ServiceLoader`
  to locate a default `RegistryHttpClient` instance,
  to `RegistryHttpClientServiceLoader`

There's also a couple of small changes to Maven POMs:
- use correct format for parent `<relativePath>` (it should point
  to a file, not a directory)
- remove a dependency on `apicurio-registry-client-vertx`
  from the `apicurio-registry-client` (it wasn't used at all,
  and was `<scope>provided</scope>` anyway)
- remove dependency management for `apicurio-registry-client-jdk`,
  such module doesn't exist

### CODE REVIEW

**Comment by Ladicek** [comment]
Hi @carlesarnal, these are the changes I mentioned yesterday on our meeting. As you can see, none of it is important -- it's just an attempt to improve internal consistency of the HTTP client SPI.

File: rest-client/client-common/pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
         <artifactId>apicurio-registry</artifactId>
         <groupId>io.apicurio</groupId>
         <version>2.1.0-SNAPSHOT</version>
-        <relativePath>../../</relativePath>
+        <relativePath>../../pom.xml</relativePath>
```
Comment (carlesarnal) [inline rest-client/client-common/pom.xml:25]: Good catch, thx

**Review by carlesarnal — APPROVED** [review]
All good, thanks for the changes @Ladicek

**Comment by Ladicek** [comment]
Cool! Just FYI, here's how the Quarkus integration looks like: https://github.com/Ladicek/quarkus/commit/29c46244a0e8aae510d1ab6ae64b03a78c26f858 If you ignore the copy of the SPI implementation, it's literally more lines of `pom.xml` than actual Java code :-)

**Comment by carlesarnal** [comment]
Thanks for the pointer, I'll take a look!

**Pull request merged**